flow/layers/physical_shape_layer_unittests.cc

```diff
@@ -200,6 +200,91 @@ TEST_F(PhysicalShapeLayerTest, ElevationComplex) {
                0, MockCanvas::DrawPathData{layer_path, layer_paint}}}));
 }
 
+TEST_F(PhysicalShapeLayerTest, ShadowNotDependsCtm) {
+  constexpr SkScalar elevations[] = {1, 2, 3, 4, 5, 10};
+  constexpr SkScalar scales[] = {0.5, 1, 1.5, 2, 3, 5};
+  constexpr SkScalar translates[] = {0, 1, -1, 0.5, 2, 10};
+
+  SkPath path;
+  path.addRect(0, 0, 8, 8).close();
+
+  for (SkScalar elevation : elevations) {
+    SkRect baseline_bounds = PhysicalShapeLayer::ComputeShadowBounds(
+        path, elevation, 1.0f, SkMatrix());
+    for (SkScalar scale : scales) {
+      for (SkScalar translateX : translates) {
+        for (SkScalar translateY : translates) {
+          SkMatrix ctm;
+          ctm.setScaleTranslate(scale, scale, translateX, translateY);
+          SkRect bounds = PhysicalShapeLayer::ComputeShadowBounds(
+              path, elevation, scale, ctm);
+          EXPECT_FLOAT_EQ(bounds.fLeft, baseline_bounds.fLeft);
+          EXPECT_FLOAT_EQ(bounds.fTop, baseline_bounds.fTop);
+          EXPECT_FLOAT_EQ(bounds.fRight, baseline_bounds.fRight);
+          EXPECT_FLOAT_EQ(bounds.fBottom, baseline_bounds.fBottom);
+        }
+      }
+    }
+  }
+}
+
+static bool IsRenderingEqual(const std::function<void(SkCanvas*)>& actual,
+                             const std::function<void(SkCanvas*)>& expected,
+                             const SkSize& canvas_size) {
+  sk_sp<SkSurface> actual_surface =
+      SkSurface::MakeRasterN32Premul(canvas_size.width(), canvas_size.height());
+  sk_sp<SkSurface> expected_surface =
+      SkSurface::MakeRasterN32Premul(canvas_size.width(), canvas_size.height());
+
+  actual_surface->getCanvas()->drawColor(SK_ColorWHITE);
+  expected_surface->getCanvas()->drawColor(SK_ColorWHITE);
+
+  actual(actual_surface->getCanvas());
+  expected(expected_surface->getCanvas());
+
+  SkPixmap actual_pixels;
+  EXPECT_TRUE(actual_surface->peekPixels(&actual_pixels));
+
+  SkPixmap expected_pixels;
+  EXPECT_TRUE(expected_surface->peekPixels(&expected_pixels));
+
+  int pixels_different = 0;
+  for (int y = 0; y < canvas_size.height(); y++) {
+    const uint32_t* actual_row = actual_pixels.addr32(0, y);
+    const uint32_t* expected_row = expected_pixels.addr32(0, y);
+    for (int x = 0; x < canvas_size.width(); x++) {
+      if (actual_row[x] != expected_row[x]) {
+        pixels_different++;
+      }
+    }
+  }
+  return pixels_different == 0;
```

Comment:
Perhaps use `return pixels_different` (with a different function name) so that the caller can `EXPECT_EQ(..., 0)` and we'll see just how many pixels failed in the error logs.